core/test/utils/matrix_generator.hpp

```diff
@@ -129,6 +129,61 @@ std::unique_ptr<MatrixType> generate_random_matrix(
 }
 
 
+/**
+ * Generates a random sparsity matrix.
+ *
+ * @tparam MatrixType  type of matrix to generate (matrix::Dense must implement
+ *                     the interface `ConvertibleTo<MatrixType>`)
+ * @tparam NonzeroDistribution  type of nonzero distribution
+ * @tparam Engine  type of random engine
+ * @tparam MatrixArgs  the arguments from the matrix to be forwarded.
+ *
+ * @param num_rows  number of rows
+ * @param num_cols  number of columns
+ * @param nonzero_dist  distribution of nonzeros per row
+ * @param engine  a random engine
+ * @param exec  executor where the matrix should be allocated
+ * @param args  additional arguments for the matrix constructor
+ */
+template <typename MatrixType = matrix::Dense<>, typename NonzeroDistribution,
+          typename Engine, typename... MatrixArgs>
+std::unique_ptr<MatrixType> generate_random_sparsity_matrix(
+    size_type num_rows, size_type num_cols, NonzeroDistribution &&nonzero_dist,
+    Engine &&engine, std::shared_ptr<const Executor> exec,
+    MatrixArgs &&... args)
+{
+    using value_type = typename MatrixType::value_type;
+    using index_type = typename MatrixType::index_type;
+    using std::begin;
+    using std::end;
+
+    matrix_data<value_type, index_type> data{gko::dim<2>{num_rows, num_cols},
+                                             {}};
+
+    std::vector<size_type> col_idx(num_cols);
+    std::iota(begin(col_idx), end(col_idx), size_type(0));
+    value_type one = 1.0;
+    for (size_type row = 0; row < num_rows; ++row) {
+        // randomly generate number of nonzeros in this row
+        auto nnz_in_row = static_cast<size_type>(nonzero_dist(engine));
+        nnz_in_row = std::max(size_type(0), std::min(nnz_in_row, num_cols));
+        // select a subset of `nnz_in_row` column indexes, and fill these
+        // locations with random values
+        std::shuffle(begin(col_idx), end(col_idx), engine);
+        std::for_each(
+            begin(col_idx), begin(col_idx) + nnz_in_row,
+            [&](size_type col) { data.nonzeros.emplace_back(row, col, one); });
+    }
+
+    data.ensure_row_major_order();
+
+    // convert to the correct matrix type
+    auto result = MatrixType::create(exec, std::forward<MatrixArgs>(args)...);
+    result->read(data);
+    return result;
+}
```

Comment:
Since the code is the same as the one for `generate_random_matrix`, except we want to pass `one` as a value, maybe we should have a function which does both and takes as parameter how we create the value (either `get_rand_value` or `one`, or anything else if we ever need it).

Reply:
That would also save you the trouble of writing a dedicated test for this.

Reply:
As this is a randomly generated matrix, I dont think it possible to explicitly test the generated matrix as such, but what is done is that the limits are checked to see if they are within the deviation. I could add a parameter which takes in a value to set it to something other than one as well, with one as default. I think leaving the `get_rand_value` as a separate function would be better.

Reply:
@tcojean , is this okay with you, if not, I can change try to change it to one function.